p5js/FaceApi/FaceApi_Image_Landmarks/sketch.js

```diff
@@ -0,0 +1,116 @@
+let faceapi;
+let img;
+let detections;
+
+// by default all options are set to true
+const detection_options = {
+    withFaceLandmarks: true,
+    withFaceExpressions: false,
+    withFaceDescriptors: false,
+}
+
+function preload(){
+    img = loadImage('assets/frida.jpg')
+}
+
+function setup() {
+    createCanvas(200, 200);
+    img.resize(width, height);
+
+    faceapi = ml5.faceApi(detection_options, modelReady)
+    textAlign(RIGHT);
+}
+
+function modelReady() {
+    console.log('ready!')
+    console.log(faceapi)
+    faceapi.detectSingle(img, gotResults)
+
+}
+
+function gotResults(err, result) {
+    if (err) {
+        console.log(err)
+        return
+    }
+    // console.log(result)
+    detections = result;
+
+    // background(220);
+    background(255);
+    image(img, 0,0, width, height)
+    if (detections) {
+        // console.log(detections)
+        drawBox(detections)
+        drawLandmarks(detections)
+    }
+}
+
+function drawBox(detections){
+    const {alignedRect} = detections;
+    const {_x, _y, _width, _height} = alignedRect._box;
+    noFill();
+    stroke(255, 0, 0)
+    strokeWeight(2)
+    rect(_x, _y, _width, _height)
+}
+
+function drawLandmarks(detections){
+    const {landmarks} = detections;
+    const mouth = landmarks.getMouth();
+    const nose = landmarks.getNose();
+    const leftEye = landmarks.getLeftEye();
+    const leftEyeBrow = landmarks.getLeftEyeBrow();
+    const rightEye = landmarks.getRightEye();
+    const rightEyeBrow = landmarks.getRightEyeBrow();
+    
+    noFill();
+    stroke(0, 0, 0)
+    strokeWeight(2)
+    
+    push()
+    // mouth
+    beginShape();
+    mouth.forEach(item => {
```

Comment:
I tend to use `for...of` in my beginner p5 teaching, not sure what will ultimately be more consistent with p5 once more ES6 syntax gets adopted.

Reply:
Here I just used a regular for loop for now.